lib/fluent/plugin/out_mongo.rb

```diff
@@ -198,11 +202,27 @@ def client(database = @database)
     def collect_records(chunk)
       records = []
       time_key = @inject_config.time_key if @inject_config
+      date_key = @parse_date_key if @parse_date_key
+
+
       tag = chunk.metadata.tag
       chunk.msgpack_each {|time, record|
         record = inject_values_to_record(tag, time, record)
         # MongoDB uses BSON's Date for time.
         record[time_key] = Time.at(time || record[time_key]) if time_key
+
+        if date_key
+          begin
+            if record[date_key].is_a? Integer
+              record[date_key] = Time.at(record[date_key] / 1000.0)
+            else
+              record[date_key] = Time.parse record[date_key] if record[date_key]
+            end
+          rescue ArgumentError
+            log.error "Failed to parse field #{@parse_date_key}. Expected valid date string value: #{record[@parse_date_key]}"
```

Comment:
*string value* is inappropriate. Should we use *integer/string value* instead?
And error log level is too high for me. I feel warn log level is enough....
```suggestion
            log.warn "Failed to parse field #{@parse_date_key}. Expected valid date integer/string value: #{record[@parse_date_key]}"
```